AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -650,15 +655,23 @@ - (ASLayout *)measureWithSizeRange:(ASSizeRange)constrainedSize
     if ([[self class] usesImplicitHierarchyManagement]) {
       if (_layout) {
         NSIndexSet *insertions, *deletions;
-        [_layout.sublayouts asdk_diffWithArray:newLayout.sublayouts insertions:&insertions deletions:&deletions compareBlock:^BOOL(ASLayout *lhs, ASLayout *rhs) {
+        [_layout.immediateSublayouts asdk_diffWithArray:newLayout.immediateSublayouts
+                                               insertions:&insertions
+                                                deletions:&deletions
+                                             compareBlock:^BOOL(ASLayout *lhs, ASLayout *rhs) {
           return ASObjectIsEqual(lhs.layoutableObject, rhs.layoutableObject);
         }];
-        _insertedSubnodes = [self _filterNodesInLayouts:newLayout.sublayouts withIndexes:insertions];
-        _deletedSubnodes = [self _filterNodesInLayouts:_layout.sublayouts withIndexes:deletions];
+        _insertedSubnodes = [self _nodesInLayout:newLayout atIndexes:insertions];
+        _deletedSubnodes = [self _nodesInLayout:_layout atIndexes:deletions filterNodes:_insertedSubnodes];
       } else {
-        NSIndexSet *indexes = [NSIndexSet indexSetWithIndexesInRange:NSMakeRange(0, [newLayout.sublayouts count])];
-        _insertedSubnodes = [self _filterNodesInLayouts:newLayout.sublayouts withIndexes:indexes];
-        _deletedSubnodes = @[];
+        NSIndexSet *indexes = [NSIndexSet indexSetWithIndexesInRange:NSMakeRange(0, [newLayout.immediateSublayouts count])];
+        _insertedSubnodes = [self _nodesInLayout:newLayout atIndexes:indexes];
```

Comment:
Does this method _nodesInLayout only return nodes that aren't currently subnodes?  Can you think of an intuitive naming tweak to communicate that, maybe the variable name actually?